PuzzleSlider/src/Engine/Engine.java

```diff
@@ -20,6 +20,7 @@ public Engine(int size) throws InvalidArgumentException {
 		mGameData = new GameData(size);
 		mGameData.initialize();
 		shuffleNewGame(1000);
+		mGameData.initVariables();
```

Comment:
Maybe just misused function name. Should be more like "initCounters()". The problem here is, we can't initialize them in "initialize()" function because the "suffleNewGame" function is then interfering with counters i.e. they are counting moves and time during shuffling new game.